modules/reactor/src/meshgenerators/AzimuthalBlockSplitGenerator.C

```diff
@@ -270,6 +312,65 @@ AzimuthalBlockSplitGenerator::generate()
   const Real min_non_circular_radius =
       *std::min_element(non_circular_rad_list.begin(), non_circular_rad_list.end());
 
+  // Before re-correcting the radii, correct the mid-point of the elements that are altered
+  if (order == 2)
+    for (const auto & elem : mesh.element_ptr_range())
+    {
+      const Point & original_vertex_avg = vertex_avgs[elem->id()];
+      const Point & new_vertex_avg = elem->vertex_average();
+      if (MooseUtils::absoluteFuzzyGreaterThan((original_vertex_avg - new_vertex_avg).norm(), 0.0))
+      {
+        if (elem->type() == TRI6 || elem->type() == TRI7)
+        {
+          *elem->node_ptr(3) = midPointCorrector(
+              *elem->node_ptr(0), *elem->node_ptr(1), max_circular_radius, rad_tol);
+          *elem->node_ptr(4) = midPointCorrector(
+              *elem->node_ptr(1), *elem->node_ptr(2), max_circular_radius, rad_tol);
+          *elem->node_ptr(5) = midPointCorrector(
+              *elem->node_ptr(2), *elem->node_ptr(0), max_circular_radius, rad_tol);
+          if (elem->type() == TRI7)
+            *elem->node_ptr(6) =
+                (*elem->node_ptr(0) + *elem->node_ptr(1) + *elem->node_ptr(2)) / 3.0;
```

Comment:
If you take an average of all 6 other nodes you'll get a nicer mapping Jacobian in the curved-edge case.